src/payload.hpp

```diff
@@ -23,16 +23,53 @@ using std::string;
 using std::map;
 using json = nlohmann::json;
 
+/**
+ * @brief Snowplow event payload with event properties.
+ */
 class Payload {
 private:
   map<string, string> m_pairs;
 
 public:
   ~Payload();
+
+  /**
+   * @brief Add a property to the payload.
+   * 
+   * @param key Property key
+   * @param value Property value
+   */
   void add(const string & key, const string & value);
+
+  /**
+   * @brief Add a map of properties to the payload.
+   * 
+   * @param pairs Key-value pairs
+   */
   void add_map(map<string, string> pairs);
+
+  /**
+   * @brief Add properties from another payload.
+   * 
+   * @param p Payload to add values from
+   */
   void add_payload(Payload p);
+
+  /**
+   * @brief Add self-describing JSON data to the payload.
+   * 
+   * @param j Self-describing JSON
+   * @param base64Encode Should the data be base64 encoded
+   * @param encoded Key for encoded data
+   * @param not_encoded Key for not-encoded data
+   */
   void add_json(json j, bool base64Encode, const string & encoded, const string & not_encoded);
+
+  /**
+   * @brief Get the payload key-value pairs.
+   * 
+   * @return map<string, string> Payload as key-value pairs
```

Comment:
Not sure it's a mistake or desired. Do you want to add the type in the `return` description?

Reply:
That was a mistake, not sure how it ended up there, thanks!